scipy/optimize/_nnls.py

```diff
@@ -1,11 +1,18 @@
 import numpy as np
 from ._cython_nnls import _nnls
+from scipy._lib.deprecation import _deprecate_positional_args, _NoValue
 
 
 __all__ = ['nnls']
 
+_nnls_dep_message = ("The 'atol' parameter is deprecated and will be removed in "
+                     "SciPy 1.18.0. It is not used in the implementation.")
 
-def nnls(A, b, maxiter=None, *, atol=None):
+@_deprecate_positional_args(version='1.18.0',
+                           deprecated_args={'atol'},
+                           custom_message=_nnls_dep_message)
+
+def nnls(A, b, maxiter=None, *, atol=_NoValue):
```

Comment:
```suggestion
@_deprecate_positional_args(version='1.18.0',
                            deprecated_args={'atol'},
                            custom_message=_nnls_dep_message)
def nnls(A, b, *, maxiter=None, atol=_NoValue):
```
If we're going to update the signature let's move   the `*` to the ideal location

Reply:
Thank you @j-bowhay